rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -145,12 +146,21 @@ def __init__(
         self.chain_modules_prefix_length = len(self.chain_modules_root)
         self.dbevmtx = DBEvmTx(self.database)
         self.dbevents = DBHistoryEvents(self.database)
-        self.base = BaseDecoderTools(
-            database=database,
-            evm_inquirer=self.evm_inquirer,
-            is_non_conformant_erc721_fn=self._is_non_conformant_erc721,
-            address_is_exchange_fn=self._address_is_exchange,
-        )
+        self.base: BaseDecoderTools
+        if isinstance(self.evm_inquirer, EvmNodeInquirerWithDSProxy):
```

Comment:
Since you have a different class `EVMTransactionDecoderWithDSProxy` you do not need this if in here as this goes again against the entire point of using classes.

Pass the BaseDecoder tools as an argument and initialize the proper type at the appropriate chain. So ethereum would use BaseDecoderToolsWithDSProxy, Polygon without